engine/src/main/java/org/terasology/engine/GameEngine.java

```diff
@@ -28,6 +28,23 @@
 public interface GameEngine extends AutoCloseable {
 
     /**
+     * @return The current, fine-grained status of the engine.
+     */
+    EngineStatus getStatus();
```

Comment:
P.S. in the scenario described in the comment above changeState() could be renamed to changeGameState().